lib/cuda/cudaDeviceInterface.cpp

```diff
@@ -87,3 +90,230 @@ void cudaDeviceInterface::async_copy_gpu_to_host(void* dst, void* src, size_t le
     CHECK_CUDA_ERROR(cudaEventCreate(&copy_end_event));
     CHECK_CUDA_ERROR(cudaEventRecord(copy_end_event, getStream(cuda_stream_id)));
 }
+
+void cudaDeviceInterface::build(const std::string& kernel_filename,
+                                const std::vector<std::string>& kernel_names,
```

Comment:
Given that it's easier now to accidentally call `build()` multiple times if someone forgets the `inst == 0` check, we should likely check if a kernel with that name already exists and throw an error if it does.    And genetically it's likely good to have that check in case two kernels end up with the same name.

Reply:
Good idea, done!